packages/components-js/tests/e2e/puppeteer/specs/wordmark.e2e.ts

```diff
@@ -0,0 +1,222 @@
+import type { Page } from 'puppeteer';
+import {
+  addEventListener,
+  expectA11yToMatchSnapshot,
+  getActiveElementId,
+  getActiveElementTagNameInShadowRoot,
+  getAttribute,
+  getEventSummary,
+  getLifecycleStatus,
+  selectNode,
+  setContentWithDesignSystem,
+  setProperty,
+  waitForStencilLifecycle,
+} from '../helpers';
+
+let page: Page;
+beforeEach(async () => (page = await browser.newPage()));
+afterEach(async () => await page.close());
+
+const getHost = () => selectNode(page, 'p-wordmark');
+const getLink = () => selectNode(page, 'p-wordmark >>> a');
+const getImage = () => selectNode(page, 'p-wordmark >>> img');
+
+const initWordmark = (opts?: {
+  href?: boolean;
+  isWrapped?: boolean;
+  hasFocusableElementBefore?: boolean;
+  hasFocusableElementAfter?: boolean;
+}): Promise<void> => {
+  const { href = false } = opts || {};
+  const { isWrapped = false } = opts || {};
+  const { hasFocusableElementBefore = false } = opts || {};
+  const { hasFocusableElementAfter = false } = opts || {};
+
+  const focusableElementBefore = hasFocusableElementBefore ? `<a href="#" id="before">before</a>` : '';
+  const focusableElementAfter = hasFocusableElementAfter ? `<a href="#" id="after">after</a>` : '';
+  const markup = `<p-wordmark ${href ? 'href="about:blank#" ' : ''} id="my-wordmark"></p-wordmark>`;
+
+  return setContentWithDesignSystem(
+    page,
+    isWrapped ? `<div>${focusableElementBefore}${markup}${focusableElementAfter}</div>` : markup
+  );
+};
+
+describe('with link', () => {
+  it('should render <a> tag when href prop is defined', async () => {
+    await initWordmark();
+    const host = await getHost();
+
+    expect(await getLink()).toBe(null);
+
+    await setProperty(host, 'href', '#some-link');
+    await waitForStencilLifecycle(page);
+
+    expect(await getLink()).not.toBe(null);
+  });
+
+  it('should sync href and target prop when changed programmatically', async () => {
+    await initWordmark({ href: true });
+    const host = await getHost();
+    const link = await getLink();
+
+    expect(await getAttribute(link, 'href')).toBe('about:blank#');
+    expect(await getAttribute(link, 'target')).toBe('_self');
+
+    await setProperty(host, 'href', '#some-link');
+    await setProperty(host, 'target', '_blank');
+    await waitForStencilLifecycle(page);
+
+    expect(await getAttribute(link, 'href')).toBe('#some-link');
```

Comment:
why are we testing stencil's prop change detection?

Reply:
I also implemented similar tests we had for p-marque - if the tests are useless I can remove them? (for both?)